### PR TITLE
Gremlin Javascript, Gremlint invalid Character Error fixes for Windows Build

### PR DESCRIPTION
**Pull request by L0Lmaker**
*Issue #, if available:*
https://bitquill.atlassian.net/browse/AN-1146

*Description of changes:*
Fixed issues where running `mvn clean install` on Windows causes:
1.
```
error  Expected linebreaks to be 'LF' but found 'CRLF'  linebreak-style
error  Delete `␍`                                       prettier/prettier
```
2.
```
Script1.groovy: 3: unexpected char: '\' @ line 3, column 24.
```
3.
```
> gremlin@3.5.4-alpha1 test
> ./node_modules/mocha/bin/mocha test/unit test/integration -t 5000

'.' is not recognized as an internal or external command,
operable program or batch file.
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: gremlin-javascript/pom.xml [hunk 1]
```diff
@@ -116,10 +116,11 @@ limitations under the License.
...
                                 <script>
-def mavenVersion = "${project.version}"
-def versionForJs = mavenVersion.replace("-SNAPSHOT", "-alpha1")
-def file = new File("${project.basedir}/src/main/javascript/gremlin-javascript/package.json")
-file.write(file.getText("UTF-8").replaceFirst(/"version": "(.*)",/, "\"version\": \"" + versionForJs + "\","))
+                                    def mavenVersion = "${project.version}"
+                                    def versionForJs = mavenVersion.replace("-SNAPSHOT", "-alpha1")
+                                    def path = new File(".").absolutePath.replace("\\.", "")
+                                    def file = new File(path + "/gremlin-javascript/src/main/javascript/gremlin-javascript/package.json")
+                                    file.write(file.getText("UTF-8").replaceFirst(/"version": "(.*)",/, "\"version\": \"" + versionForJs + "\","))
```
Comment (lyndonbauto) [inline gremlin-javascript/pom.xml]: Is the tabbing necessary?

Comment (L0Lmaker) [reply gremlin-javascript/pom.xml]: Not necessary, but is that not the XML convention?

File: .github/workflows/build-test.yml [hunk 2]
```diff
@@ -13,7 +13,7 @@ jobs:
           java-version: '11'
           distribution: 'temurin'
       - name: Build with Maven
-        run: mvn clean install -DskipTests -Dci --batch-mode -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn
+        run: mvn clean install -DskipTests -Dci --batch-mode -D"org.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener"="warn"
```
Comment (lyndonbauto) [inline .github/workflows/build-test.yml]: was this not working?

File: .github/workflows/build-test.yml [hunk 3]
```diff
@@ -126,7 +126,10 @@ jobs:
     name: javascript
     timeout-minutes: 15
     needs: smoke
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ ubuntu-latest, windows-latest]
```
Comment (lyndonbauto) [inline .github/workflows/build-test.yml]:
```suggestion
        os: [ubuntu-latest, windows-latest]
```

File: .github/workflows/build-test.yml (hunk 2 above)
Comment (L0Lmaker) [reply .github/workflows/build-test.yml]: Do we need the smoke test also to have a Windows run? In windows the `"` is required for it to run. Currently the smoke test only runs on Linux so we can remove this change and be fine.

Comment (lyndonbauto) [reply .github/workflows/build-test.yml]: Sure that sounds good.

**Review by lyndonbauto — APPROVED** [review]
LGTM

File: gremlin-javascript/pom.xml [hunk 4]
```diff
@@ -116,10 +116,11 @@ limitations under the License.
                         <configuration>
                             <scripts>
                                 <script>
-def mavenVersion = "${project.version}"
-def versionForJs = mavenVersion.replace("-SNAPSHOT", "-alpha1")
-def file = new File("${project.basedir}/src/main/javascript/gremlin-javascript/package.json")
-file.write(file.getText("UTF-8").replaceFirst(/"version": "(.*)",/, "\"version\": \"" + versionForJs + "\","))
+                                    def mavenVersion = "${project.version}"
+                                    def versionForJs = mavenVersion.replace("-SNAPSHOT", "-alpha1")
+                                    def platformAgnosticBaseDirPath = new File(".").absolutePath.replace("\\.", "").replace("\\", "/")
```
Comment (simonz-bq) [inline gremlin-javascript/pom.xml]: I'm worried about the new File(".") being not unique enough, but this is more of a worry than a blocker.

File: gremlin-server/src/test/scripts/test-server-start.groovy [hunk 5]
```diff
@@ -48,15 +48,22 @@ if (testTransactions) {
...
-settings.graphs.modern = gremlinServerDir + "/src/test/scripts/tinkergraph-empty.properties"
-settings.graphs.crew = gremlinServerDir + "/src/test/scripts/tinkergraph-empty.properties"
-settings.graphs.grateful = gremlinServerDir + "/src/test/scripts/tinkergraph-empty.properties"
-settings.graphs.sink = gremlinServerDir + "/src/test/scripts/tinkergraph-empty.properties"
-if (testTransactions) settings.graphs.tx = gremlinServerDir + "/src/test/scripts/neo4j-empty.properties"
-settings.scriptEngines["gremlin-groovy"].plugins["org.apache.tinkerpop.gremlin.jsr223.ScriptFileGremlinPlugin"].files = [gremlinServerDir + "/src/test/scripts/generate-all.groovy"]
+def platformAgnosticBaseDirPath = new File(".this-definitely-does-not-exist").absolutePath
+        .replace("\\.this-definitely-does-not-exist", "") // Windows
+        .replace("\\", "/") // Windows
+        .replace("/.this-definitely-does-not-exist","") // Unix
```
Comment (simonz-bq) [inline gremlin-server/src/test/scripts/test-server-start.groovy:54]: Can use this name for above

File: gremlint/pom.xml [hunk 6]
```diff
@@ -72,7 +72,8 @@ limitations under the License.
                                 <script>
                                     def mavenVersion = "${project.version}"
                                     def versionForJs = mavenVersion.replace("-SNAPSHOT", "-alpha1")
-                                    def file = new File("${project.basedir}/package.json")
+                                    def path = new File(".").absolutePath.replace("\\.", "").replace("\\","/")
```
Comment (simonz-bq) [inline gremlint/pom.xml]: And here too.

**Pull request merged**